cpp/WriterPropertiesBuilder.cpp

```diff
@@ -209,4 +209,21 @@ extern "C"
 	{
 		TRYCATCH(builder->disable_page_checksum();)
 	}
+
+	PARQUETSHARP_EXPORT ExceptionInfo* WriterPropertiesBuilder_Sorting_Columns(WriterProperties::Builder* builder, int32_t* column_indices, bool* descending, bool* nulls_first, int num_columns)
+	{
+		std::vector<parquet::SortingColumn> sorting_columns;
+		
+		// Create SortingColumn objects for each column
+		for (int i = 0; i < num_columns; ++i)
+		{
+			parquet::SortingColumn column;
+			column.column_idx = column_indices[i];
+			column.descending = descending[i];
+			column.nulls_first = nulls_first[i];
+			sorting_columns.push_back(column);
```

Comment:
To avoid copying the stack variable, we can use:
```suggestion
			sorting_columns.emplace_back(column);
```

Reply:
> I'm not a C++ expert, but my understanding is using emplace_back like this doesn't help as we've already created the column, which will need to be copied.

Yeah, you are right. my suggestion wouldn't change anything.